cmake/modules/SwiftSharedCMakeConfig.cmake

```diff
@@ -78,11 +78,13 @@ macro(swift_common_standalone_build_config_llvm product is_cross_compiling)
   include(AddSwiftTableGen) # This imports TableGen from LLVM.
   include(HandleLLVMOptions)
 
-  # HACK: this ugly tweaking is to prevent the propagation of the flag from LLVM
-  # into swift.  The use of this flag pollutes all targets, and we are not able
-  # to remove it on a per-target basis which breaks cross-compilation.
-  # Therefore, we always remove it here and then selectively re-add it to the
-  # link flags.
+  # HACK: Windows doesn't support -z,defs as a linker flag. 
+  # Normally, LLVM would only add it as an option for non-Windows systems;
+  # however, due to the custom scheme Swift uses for cross-compilation, the 
+  # CMAKE_SHARED_LINKER_FLAGS are determined based on the host system and 
+  # then applied to all targets. 
+  # To work around this, we unconditionally remove the flag here and then 
+  # selectively add it to the per-target link flags in AddSwift.cmake.
```

Comment:
Nice explanation! It is exactly what I was looking for. My biggest fear in general with cmake changes is that we lose

3 last knits:

1. Can you fix the formatting on the text so it is 80 columns or proper paragraphs? I.e. I would do this:

```
# HACK: Windows doesn't support -z,defs as a linker flag.
#
# Normally, LLVM ...
```

2. Can you say the function in AddSwift.cmake where it is (add_swift_library, I think it is?).
3. I forgot if it is there, but in add_swift_library can you add a similar comment to this that points back at this code in words like you pointed at it in words here.

Reply:
Erm.

1. knits => nits. 
2. What I meant to say is that my biggest fear with cmake changes is forgetting why hacks were added. That is why I am asking for so much here as such things have happened in the past.

Reply:
Does that look okay now? I've been editing on the GitHub web interface so gauging formatting is somewhat difficult; I don't think it exposes column width, for instance. I've also updated the comment in AddSwift.cmake so it's hopefully a little clearer.